ROOTFS/usr/bin/waggle-network-watchdog

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+from pathlib import Path
+import subprocess
+import time
+import logging
+from glob import glob
+
+logging.basicConfig(level=logging.INFO)
+
+last_connection_time = 0
+reset_timeout = 0
+
+
+def update_systemd_watchdog():
+    try:
+        subprocess.check_call(["systemd-notify", "WATCHDOG=1"])
+    except subprocess.CalledProcessError:
+        logging.warning("skipping reset of systemd watchdog")
+
+
+def time_now():
+    return time.monotonic()
+
+
+def seconds_since(start):
+    return time.monotonic() - start
+
+
+def ssh_connection_ok():
+    try:
+        return (
+            "beehive:20022"
+            in subprocess.check_output(["ss", "-tr", "state", "established"]).decode()
+        )
+    except Exception:
+        return False
+
+
+# NOTE Revisit how much of the network stack we should restart. For now, I want to cover all
+# cases of wifi and modems and ssh tunnel issues.
+def restart_network_services():
+    logging.warning("restarting network services")
+
+    # ensure proper ownership of ports, ttyACM* for Modem
+    ports = glob("/dev/ttyACM*")
+    subprocess.run(["chown", "root:root"] + ports)
+    subprocess.run(["chmod", "660"] + ports)
+
+    # restart network services
+    subprocess.run(
+        [
+            "systemctl",
+            "restart",
+            "NetworkManager",
+            "ModemManager",
+            "waggle-reverse-tunnel",
+        ]
+    )
+
+
+def reboot_os():
+    subprocess.run(["systemctl", "--force", "reboot"])
+    reset_timeout = time_now()
+
+
+last_connection_time = time_now()
+reset_timeout = time_now()
+
+# Recovery actions table [time (s), recovery function]
+# restart networking stack after 15, 20 and 25 of no beehive connectivity
+# reboot after 30 mins of no beehive connectivity
+recovery_actions = [
+    (1800, reboot_os),
+    (1500, restart_network_services),
+    (1200, restart_network_services),
+    (900, restart_network_services),
+]
+
+while True:
+    update_systemd_watchdog()
+
+    if ssh_connection_ok():
+        logging.info("connection ok")
+        last_connection_time = time_now()
+        reset_timeout = time_now()
+    else:
+        logging.warning(
+            "no connection for %ss", int(seconds_since(last_connection_time))
+        )
+
+    for action in recovery_actions:
+        if seconds_since(reset_timeout) >= action[0]:
+            action[1]()
+            recovery_actions.remove(action)
```

Comment:
I don't think we want this....  why did you have this here?

This will cause the action to be forever removed from the action list.  If we execute a recovery (@ 900s) that fixes the system then the system fails again the @ 900s recovery action no longer exists.